pkg/apis/core/validation/validation_test.go

```diff
@@ -25002,6 +25003,120 @@ func TestValidateHostUsers(t *testing.T) {
 	}
 }
 
+func TestValidatePodHostName(t *testing.T) {
+	trueVal := true
+	falseVal := false
+	tests := []struct {
+		name         string
+		spec         core.PodSpec
+		expectedErrs field.ErrorList
+	}{
+		{
+			name: "SetHostnameAsFQDN=true and HostnameOverride is set should error",
+			spec: core.PodSpec{
+				SetHostnameAsFQDN: &trueVal,
+				HostnameOverride:  ptr.To("custom-host"),
+			},
+			expectedErrs: field.ErrorList{
+				field.Forbidden(field.NewPath("spec").Child("hostnameOverride"), "may not be specified when setHostnameAsFQDN is true"),
```

Comment:
Unless you add one of the `ByDetail<...>()` calls to your matcher, the detail string here is just noise.  I would add `ByDetailSubstring()` and then just say "setHostnameAsFQDN" here.

Reply:
Small trick on fieldpaths -- if you don't need to modify or analyze them piece-by-piece, you can just say `field.NewPath("spec.hostnameOverride")`